tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -0,0 +1,331 @@
+            self.assertAlmostEqual(percent_equal, 1.0, delta=.10,
+                                   msg='bucket: %s' % bucket)
+
+    def test_variant_returns_none_if_out_of_time_window(self):
+        valid_cfg = get_simple_config()
+        experiment_valid = parse_experiment(valid_cfg)
+        experiment_valid._choose_variant = choose_variants_override
+
+        expired_cfg = get_simple_config()
+        expired_cfg['stop_ts'] = time.time() - FIVE_DAYS
+        experiment_expired = parse_experiment(expired_cfg)
+
+        experiment_not_started_cfg = get_simple_config()
+        experiment_not_started_cfg['start_ts'] = time.time() + FIVE_DAYS
+        experiment_not_started = parse_experiment(experiment_not_started_cfg)
+
+        variant_valid = experiment_valid.variant(user_id="t2_1")
+        variant_expired = experiment_expired.variant(user_id="t2_1")
+        variant_not_started = experiment_not_started.variant(user_id="t2_1")
+
+        self.assertIsNot(variant_valid, None)
+        self.assertIs(variant_expired, None)
+        self.assertIs(variant_not_started, None)
+
+    def test_no_bucket_val(self):
+        experiment = create_simple_experiment()
+        no_user_id_provided_variant = experiment.variant(not_user_id="t2_1")
+        none_user_id_provided_variant = experiment.variant(not_user_id=None)
+
+        self.assertIs(no_user_id_provided_variant, None)
+        self.assertIs(none_user_id_provided_variant, None)
+
+    def test_experiment_disabled(self):
+        experiments_cfg = get_simple_config()
+        experiments_cfg['experiment']['enabled'] = False
+
+        experiment = parse_experiment(experiments_cfg)
+
+        variant = experiment.variant(user_id="t2_1")
+        self.assertIs(variant, None)
+
+    def test_bucket_val(self):
+        cfg = {"id": 1,
+            "name": "test_experiment",
+            "owner": "test",
+            "type": "single_variant",
+            "version": "1",
+            "start_ts": time.time() - THIRTY_DAYS,
+            "stop_ts": time.time() + THIRTY_DAYS,
+            "enabled": True,
+            "experiment": {
+                "variants": [
+                    {
+                        "name":"variant_1",
+                        "size":0.5,
+                    },
+                    {
+                        "name":"variant_2",
+                        "size":0.5,
+                    },
+                ],
+                "experiment_version":1,
+                "bucket_val":"new_bucket_val",
+            }
+        }
+
+        experiment = parse_experiment(cfg)
+
+        experiment._choose_variant = choose_variants_override
+
+        variant_default_bucket_val = experiment.variant(user_id="t2_1")
+        variant_new_bucket_val = experiment.variant(new_bucket_val="some_value")
+
+        self.assertIs(variant_default_bucket_val, None)
+        self.assertIsNot(variant_new_bucket_val, None)
+
+    def test_change_shuffle_version_changes_bucketing(self):
```

Comment:
You might want to add the `skipIf` decorator used in other tests to this one since it's running a lot of bucket checks.

Reply:
it's only running a lot of bucketing checks if users keep getting bucketed into the same buckets, which shouldn't happen. It will end the loop as soon as a user winds up in two different buckets, which should happen very quickly (likely the first or second iteration of the loop.

I don't like skipping tests locally where not absolutely necessary.